src/tests/testHttpRequest.cc

```diff
@@ -121,36 +123,48 @@ TestHttpRequest::testIPv6HostColonBug()
 {
     HttpRequest *aRequest = nullptr;
 
-    /* valid IPv6 address without port */
+    /* bracketed IPv6 address without port */
     SBuf url("http://[2000:800::45]/foo");
     const auto mx = MasterXaction::MakePortless<XactionInitiator::initHtcp>();
     aRequest = HttpRequest::FromUrl(url, mx, Http::METHOD_GET);
-    AnyP::KnownPort expected_port = 80;
-    CPPUNIT_ASSERT_EQUAL(expected_port, *aRequest->url.port());
-    CPPUNIT_ASSERT(aRequest->method == Http::METHOD_GET);
-    CPPUNIT_ASSERT_EQUAL(String("[2000:800::45]"), String(aRequest->url.host()));
-    CPPUNIT_ASSERT_EQUAL(SBuf("/foo"), aRequest->url.path());
-    CPPUNIT_ASSERT_EQUAL(AnyP::PROTO_HTTP, static_cast<AnyP::ProtocolType>(aRequest->url.getScheme()));
-
-    /* valid IPv6 address with port */
+    if (Ip::EnableIpv6) {
+        CPPUNIT_ASSERT(aRequest != nullptr);
+        CPPUNIT_ASSERT_EQUAL(AnyP::KnownPort(80), *aRequest->url.port());
+        CPPUNIT_ASSERT(aRequest->method == Http::METHOD_GET);
+        CPPUNIT_ASSERT_EQUAL(String("[2000:800::45]"), String(aRequest->url.host()));
+        CPPUNIT_ASSERT_EQUAL(SBuf("/foo"), aRequest->url.path());
+        CPPUNIT_ASSERT_EQUAL(AnyP::PROTO_HTTP, static_cast<AnyP::ProtocolType>(aRequest->url.getScheme()));
+    } else {
+        CPPUNIT_ASSERT(!aRequest);
+    }
```

Comment:
The diff does not show this well, but we are essentially preserving the old assertions when Ip::EnableIpv6 is true (with some minor polishing touches). When Ip::EnableIpv6 is false, we simply assert that HttpRequest::FromUrl() fails. Same for the two test cases below.